### PR TITLE
[MCC-106990] Benchmark Deserialization

### PR DESCRIPTION
**Pull request by kenyamat**
@mdsol/hypermedia-team - I created a benchmarking tool.
Please review this.

### CODE REVIEW

**Comment by edandersen** [comment]
Run.bat should:
- be in the root of the repository, maybe called "benchmark.bat"
- actually compile the project with MSBuild in Release mode before running the exe

File: tools/Crichton.Representors.Benchmark/run.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.\bin\Debug\Crichton.Representors.Benchmark.exe deserializer --iterations 5 --filepath "C:\GitProjects\crichton-dotnet\tests\Crichton.Representors.Tests\Integration\TestData\Hal\AllLinkObjectProperties.json"
```
Comment (edandersen) [inline tools/Crichton.Representors.Benchmark/run.bat]: Should be a relative path here to the JSON instead of absolute

**Comment by edandersen** [comment]
Also lets put the benchmark tool project in a Solution Folder called "tools" in VS.

**Comment by kenyamat** [comment]
@edandersen Thank you for your feedback.
I modified the issues and also added README.md.

**Comment by edandersen** [comment]
@kenyamat Thanks for the updates. I checked out the branch and ran `benchmark.bat` and it successfully built the libraries and ran the benchmark.
- Please remove the "press any key to continue" Console.ReadKey as this will block if we run this in CI
- Please move the README.md to the /docs folder and add a link to it on the main project Readme,md

...then it's good to merge!

**Comment by kenyamat** [comment]
Hi @edandersen , Thank you for your review.
The code was fixed. Could you please review this again?
- Moved and renamed README.md into representor_benchmark_tool.md.
- Removed pause statement in benchmark.bat.

**Comment by edandersen** [comment]
Awesome! Merging.

**Pull request merged**